### PR TITLE
Handle pooch version 1.5.0

### LINKED ISSUE #5528: Pooch update breaks import
## Description

Pooch recently updated to version 1.5.0.  This moved the hash functions to their own module, so the current method of importing them now fails.  Further, the fallback for if pooch is not installed doesn't work because the exact error is different.

## Way to reproduce
```python
import skimage.metrics
```
Note, install pooch 1.5.0 first.

You'll see
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".../python3.8/site-packages/skimage/__init__.py", line 135, in <module>
    from .data import data_dir
  File ".../python3.8/site-packages/skimage/data/__init__.py", line 64, in <module>
    from pooch.utils import file_hash
ImportError: cannot import name 'file_hash' from 'pooch.utils' (.../python3.8/site-packages/pooch/utils.py)
```

## Version information
```python
# Paste the output of the following python commands
from __future__ import print_function
import sys; print(sys.version)
import platform; print(platform.platform())
import skimage; print("scikit-image version: {}".format(skimage.__version__))
import numpy; print("numpy version: {}".format(numpy.__version__))
```

```python
# your output here
3.8.10 (default, May  5 2021, 03:01:07)
[GCC 7.5.0]
Linux-5.4.0-80-generic-x86_64-with-glibc2.27
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".../python3.8/site-packages/skimage/__init__.py", line 135, in <module>
    from .data import data_dir
  File ".../python3.8/site-packages/skimage/data/__init__.py", line 64, in <module>
    from pooch.utils import file_hash
ImportError: cannot import name 'file_hash' from 'pooch.utils' (.../python3.8/site-packages/pooch/utils.py)
numpy version: 1.21.2
```

I'm using ```scikit-image version: 0.18.2```

## Possible Fix

One solution would be to change `skimage/data/__init__.py`, where it reads
```
    from pooch.utils import file_hash
```
to
```
    try:
        from pooch.utils import file_hash
    except ImportError:
        from pooch.hashes import file_hash
```

### PR DESCRIPTION
**Pull request by manthey**
This is compatible with pooch < 1.5.0, too.  If such older compatibility
is not desired, the dependency version could be changed and just the new
code could be kept.

## Description

Closes #5528.

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

**Review by hmaarrfk — APPROVED** [review]
Thanks! I flipped the order since I typically like to try "the new standard" instead of first trying the "old" then "falling back on the new".

**Comment by hmaarrfk** [comment]
Thank you so much for being quick on this!

File: skimage/data/__init__.py [hunk 1]
```diff
@@ -62,7 +62,12 @@
 skimage_distribution_dir = osp.join(legacy_data_dir, '..')
 
 try:
-    from pooch.utils import file_hash
+    try:
+        # Pooch >= 1.5.0
+        from pooch.hashes import file_hash
+    except ImportError:
+        # Pooch < 1.5.0
+        from pooch.utils import file_hash
```
Comment (grlee77) [inline skimage/data/__init__.py:70]:
```suggestion
    from pooch import file_hash
```
I think we can just do this instead

I looked at the pooch source and it seems this function has always been available from the main pooch namespace: 
currently: https://github.com/fatiando/pooch/blob/master/pooch/__init__.py#L12
even in v0.1: https://github.com/fatiando/pooch/blob/v0.1/pooch/__init__.py#L5

**Review by grlee77 — CHANGES_REQUESTED** [review]

File: skimage/data/__init__.py (hunk 1 above)
Comment (manthey) [reply skimage/data/__init__.py:70]: That's much more sensible.

Comment (hmaarrfk) [reply skimage/data/__init__.py:70]: thank you both for looking into this correctly.

**Comment by grlee77** [comment]
I think pooch only considers the functions/classes in the top-level `pooch` namespace to be the public API. The PR moving the `file_hash` function [refers to hashes as "a new private module"](https://github.com/fatiando/pooch/pull/244). I think we mistakenly assumed the submodules were public API since the names didn't start with an underscore, but apparently that isn't the case

**Comment by hmaarrfk** [comment]
@grlee77 how do you want to proceed to get this released? This is rather breaking for our users.

I could patch the conda-forge version, (and I might for my own channel)

**Comment by grlee77** [comment]
> @grlee77 how do you want to proceed to get this released? 

I think it is probably best to make a quick 0.18.3 release with this fix. Hopefully it should be easy using the new, automated wheel building. 

The 0.19 milestone items remaining are mostly just needing reviews on some recent small bug fix PRs and one larger float32 support PR. Hopefully we can complete those soon, but I don't think in the next day or two is likely.

**Comment by hmaarrfk** [comment]
ok. i'll hold off on making a patched version for conda-forge unless an other user complains before we have to release.

I'm not sure I can spend daylight hours pushing a sanctionned pip fix through.

**Review by grlee77 — APPROVED** [review]

**Comment by grlee77** [comment]
> I'm not sure I can spend daylight hours pushing a sanctioned pip fix through.

I posted here about potentially making a 0.18.3 release: https://skimage.zulipchat.com/#narrow/stream/181448-development/topic/v0.2E18.2E3.20release.3F

**Comment by grlee77** [comment]
@meeseeksdev backport to v0.18.x

**Pull request merged**